servicebus/data-plane/azure-servicebus/src/main/java/com/microsoft/azure/servicebus/primitives/CoreMessageSender.java

```diff
@@ -836,70 +712,54 @@ public void onEvent()
-    private void processSendWork()
-    {
-        synchronized (this.pendingSendLock)
-        {
-            if(!this.isSendLoopRunning)
-            {
+    private void processSendWork() {
+        synchronized (this.pendingSendLock) {
+            if (!this.isSendLoopRunning) {
                 this.isSendLoopRunning = true;
-            }
-            else
-            {
+            } else {
                 return;
             }
         }
 
         TRACE_LOGGER.debug("Processing pending sends to '{}'. Available link credit '{}'", this.sendPath, this.linkCredit);
-        try
-        {
-            if(!this.ensureLinkIsOpen().isDone())
-            {
+        try {
+            if (!this.ensureLinkIsOpen().isDone()) {
                 // Link recreation is pending
                 return;
             }
 
             final Sender sendLinkCurrent = this.sendLink;
             while (sendLinkCurrent != null
                     && sendLinkCurrent.getLocalState() == EndpointState.ACTIVE && sendLinkCurrent.getRemoteState() == EndpointState.ACTIVE
-                    && this.linkCredit > 0)
-            {
+                    && this.linkCredit > 0) {
                 final WeightedDeliveryTag deliveryTag;
                 final SendWorkItem<DeliveryState> sendData;
-                synchronized (this.pendingSendLock)
-                {
+                synchronized (this.pendingSendLock) {
                     deliveryTag = this.pendingSends.poll();
-                    if (deliveryTag == null)
-                    {
+                    if (deliveryTag == null) {
                         TRACE_LOGGER.debug("There are no pending sends to '{}'.", this.sendPath);
                         // Must be done inside this synchronized block
                         this.isSendLoopRunning = false;
                         break;
-                    }
-                    else
-                    {
+                    } else {
                         sendData = this.pendingSendsData.get(deliveryTag.getDeliveryTag());
-                        if(sendData == null)
-                        {
+                        if (sendData == null) {
                             TRACE_LOGGER.debug("SendData not found for this delivery. path:{}, linkName:{}, deliveryTag:{}", this.sendPath, this.sendLink.getName(), deliveryTag);
                             continue;
                         }
                     }
                 }
 
-                if (sendData.getWork() != null && sendData.getWork().isDone())
-                {
+                if (sendData.getWork() != null && sendData.getWork().isDone()) {
                     // CoreSend could enqueue Sends into PendingSends Queue and can fail the SendCompletableFuture
                     // (when It fails to schedule the ProcessSendWork on reactor Thread)
-                    this.pendingSendsData.remove(sendData);
+                    this.pendingSendsData.remove(deliveryTag.getDeliveryTag());
```

Comment:
[SpotBugs P1] Coding mistake: sendData is value in the map, it should use key class of map in remove()